web/src/test/java/com/graphhopper/application/resources/RouteResourceClientHCTest.java

```diff
@@ -379,4 +381,86 @@ public void testCustomModel() {
+                setProfile("bike");
+
+        GHResponse response = gh.route(req);
+        ResponsePath path = response.getBest();
+        assertEquals(5333, path.getDistance(), 5);
+        assertEquals(9, path.getWaypoints().size());
+
+        assertEquals(path.getTime(), path.getPathDetails().get("leg_time").stream().mapToLong(d -> (long) d.getValue()).sum(), 1);
+        assertEquals(path.getDistance(), path.getPathDetails().get("leg_distance").stream().mapToDouble(d -> (double) d.getValue()).sum(), 1);
+        assertEquals(path.getRouteWeight(), path.getPathDetails().get("leg_weight").stream().mapToDouble(d -> (double) d.getValue()).sum(), 1);
+
+        List<PointList> pointListFromInstructions = getPointListFromInstructions(path);
+        for (String detail : legDetails) {
+            List<PathDetail> pathDetails = path.getPathDetails().get(detail);
+
+            // explicitly check one of the waypoints
+            assertEquals(42.50539, path.getWaypoints().get(2).lat);
+            assertEquals(42.50539, path.getPoints().get(pathDetails.get(1).getLast()).getLat());
+            assertEquals(42.50539, path.getPoints().get(pathDetails.get(2).getFirst()).getLat());
+            // check all the waypoints
+            assertEquals(path.getWaypoints().get(0), path.getPoints().get(pathDetails.get(0).getFirst()));
+            for (int i = 1; i < path.getWaypoints().size(); ++i)
+                assertEquals(path.getWaypoints().get(i), path.getPoints().get(pathDetails.get(i - 1).getLast()));
+
+            List<PointList> pointListFromLegDetails = getPointListFromLegDetails(path, detail);
+            assertEquals(8, pointListFromLegDetails.size());
+            assertPointListsEquals(pointListFromInstructions, pointListFromLegDetails);
+        }
+    }
+
+    private List<PointList> getPointListFromInstructions(ResponsePath path) {
+        List<PointList> legs = new ArrayList<>();
+        PointList perLeg = new PointList();
+        for (Instruction instruction : path.getInstructions()) {
+            perLeg.add(instruction.getPoints());
+            if (instruction.getSign() == REACHED_VIA || instruction.getSign() == FINISH) {
+                legs.add(perLeg);
+                perLeg = new PointList();
+            } else {
+                perLeg.removeLastPoint();
+            }
+        }
+        return legs;
+    }
+
+    private List<PointList> getPointListFromLegDetails(ResponsePath path, String detail) {
+        List<PointList> legs = new ArrayList<>();
+        List<PathDetail> legDetails = path.getPathDetails().get(detail);
+        for (PathDetail legDetail : legDetails) {
+            PointList leg = new PointList(legDetail.getLast() - legDetail.getFirst() + 1, path.getPoints().is3D());
+            for (int j = legDetail.getFirst(); j <= legDetail.getLast(); j++)
+                leg.add(path.getPoints(), j);
+            legs.add(leg);
+        }
+        return legs;
+    }
```

Comment:
This is how we can use the new path details to subdivide the global point list into multiple point lists for each leg.